### ISSUE TITLE
SelectOneRadio: Accessibility Enhancement for Screenreaders - interactive controls

### ISSUE THREAD

**Issue by nichirsch12** [primefaces/primefaces#13047]
### Description

I'm experiencing some issues with the p:SelectOneRadio tag when reading my pages with the NVDA screenreader. Admittedly, this is the only screenreader I have access to, so maybe it does work with other readers. When I visit the [SelectOneRadio showcase](https://www.primefaces.org/showcase/ui/input/oneRadio.xhtml?jfwid=5d154), tab to the Page Direction group and use the arrow keys to navigate down the list, it correctly selects each option but NVDA reads the following:

```
Option1  radio button  checked  1 of 3

Option1  radio button  checked  1 of 1

Option2  radio button  not checked  2 of 3

Option2  radio button  checked  1 of 1

Option3  radio button  not checked  3 of 3

Option3  radio button  checked  1 of 1
```

### Describe the solution you would like

Interestingly, this seems to be an issue with the role='radio' attribute that is added to the parent div of the ui-radiobutton. Additionally, axe DevTools, an accessibility plugin, takes issue with this attribute as well and says "interactive controls must be not be nested". If I remove this role, axe doesn't have any more issues and NVDA reads the page correctly:

```
Option1  radio button  checked  1 of 3

Option2  radio button  checked  2 of 3

Option3  radio button  checked  3 of 3
```

I'm sure the role=radio grants us something, so obviously it should only be removed if we can verify that removing it fixes these issues and doesn't make anything else worse.

**Comment by melloware** [comment]
@nichirsch12 is it only the Page Direction example that is wrong or is it also Responsive etc. They all have `role="radio"` ?

**Comment by nichirsch12** [comment]
Sorry about the confusion. I used the Page Direction radios as an example but yes, the behavior is exhibited by each of the radio groups on that page.

**Comment by melloware** [comment]
ok agreed thos properties were not needed I agree.
